cpp/src/arrow/table.cc

```diff
@@ -619,6 +619,7 @@ TableBatchReader::TableBatchReader(const Table& table)
   for (int i = 0; i < table.num_columns(); ++i) {
     column_data_[i] = table.column(i).get();
   }
+  ARROW_DCHECK_OK(table_.ValidateFull());
```

Comment:
Please use `DCHECK(table_.Validate().ok());` so it's not validated in all builds.

`DCHECK_OK` is weird
https://github.com/apache/arrow/blob/192de026ad5c021846d89449757b0ac679436d13/cpp/src/arrow/util/logging.h#L92

Reply:
> Please use `DCHECK(table_.Validate().ok());` so it's not validated in all builds.

thanks for the suggestion. Commit amended with it.